src/EnergyPlus/SetPointManager.cc

```diff
@@ -8429,6 +8459,43 @@ namespace SetPointManager {
         }
     }
 
+    int getSPMBasedOnNode(int const NodeNum, int const SetPtType, int const SPMType, CtrlNodeType ctrlOrRefNode) {
```

Comment:
new function allows any model to find a specific SPM based on either a control node or a reference node.